crates/tx-graph2/src/connectors/cpfp.rs

```diff
@@ -47,49 +49,77 @@ impl Connector for CpfpConnector {
         panic!("P2A is not a taproot output")
     }
 
-    fn finalize_input(&self, _input: &mut bitcoin::psbt::Input, _witness: &Self::Witness) {
-        // Do nothing
+    fn finalize_input(&self, input: &mut bitcoin::psbt::Input, _witness: &Self::Witness) {
+        input.final_script_witness = Some(Witness::default());
     }
 }
 
 #[cfg(test)]
 mod tests {
-    use secp256k1::Message;
+    use bitcoin::{absolute, transaction, OutPoint, Transaction, TxOut};
+    use strata_bridge_primitives::scripts::prelude::create_tx_ins;
 
     use super::*;
-    use crate::connectors::test_utils::Signer;
-
-    struct P2ASigner;
-
-    impl Signer for P2ASigner {
-        type Connector = CpfpConnector;
-
-        fn generate() -> Self {
-            Self
-        }
-
-        fn get_connector(&self) -> Self::Connector {
-            CpfpConnector {
-                network: Network::Regtest,
-            }
-        }
-
-        fn get_connector_name(&self) -> &'static str {
-            "p2a"
-        }
-
-        fn sign_leaf(
-            &self,
-            _leaf_index: Option<usize>,
-            _sighash: Message,
-        ) -> <Self::Connector as Connector>::Witness {
-            // Return unit
-        }
-    }
+    use crate::connectors::test_utils::BitcoinNode;
 
     #[test]
     fn p2a_spend() {
-        let leaf_index = None;
-        P2ASigner::assert_connector_is_spendable(leaf_index);
+        let mut node = BitcoinNode::new();
+
+        // Create the parent transaction that funds the P2A connector.
+        // The parent transaction is v3 and has zero fees.
+        //
+        // inputs        | outputs
+        // --------------+--------------
+        // N sat: wallet | N sat: wallet
+        //               |--------------
+        //               | 0 sat: P2A
```

Comment:
```suggestion
        //               | 0 sat: P2A (CPFP)
```

It's better to label this because using P2A is one way of doing CPFP.